deployments/tools/export/lakefs_export.py

```diff
@@ -0,0 +1,117 @@
+    parser.add_argument('--prev_commit_id', metavar='previous-commit', type=str, action='store', help='a previous commit reference to export only the diff between it and the head of the branch')
+
+    args = parser.parse_args()
+    return args
+
+
+def write_status_file(success, status_file_name):
+    if success:
+        with open(status_file_name,'w') as f:
+            f.write(SUCCESS_MSG)
+    else:
+        with open('errors','r') as errors_file, open(status_file_name,'w') as status_file:
+            for line in errors_file:
+                if line.startswith('-'):
+                    status_file.write("path was missing on the source: " + line)
+                elif line.startswith('+'):
+                    status_file.write("path was missing on the destination: " + line)
+                elif line.startswith('*'):
+                    status_file.write("path was present in source and destination but different: " + line)
+                elif line.startswith('!'):
+                    status_file.write("there was an error reading or hashing the source or dest: " + line)
+    return
+
+
+def main():
+
+    # create rclone configuration file
+    create_rclone_conf_file()
+
+    args = set_args()
+
+    reference = ""
+    source = "lakefs:" + args.Repo + "/"
+    has_branch = (args.branch != None)
+    has_commit = (args.commit_id != None)
+    export_diff = (args.prev_commit_id != None)
+    if has_branch and not has_commit:
+        source += args.branch + "/"
+        reference = args.branch
+    elif not has_branch and has_commit:
+        source += args.commit_id + "/"
+        reference = args.commit_id
+    else:
+        raise Exception("Should assign branch or commit, but not both.")
+
+    if has_commit and export_diff:
+        raise Exception("Could not export diff between two commits.")
+
+    if export_diff:
+        cmd = ["rclone", "sync", source, args.Dest, "--config=rclone.conf", "--create-empty-src-dirs"]
+    else:
+        cmd = ["rclone", "copy", source, args.Dest, "--config=rclone.conf", "--create-empty-src-dirs"]
+
+    process = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # check export and create status file
+    if export_diff:
+        check_process = subprocess.run(["rclone", "check", source, args.Dest, "--config=rclone.conf", "--combined=errors"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    else:
+        # if we check the copy command we add the flag --one-way since we only want to check that the source files were copied to the destination
+        check_process = subprocess.run(["rclone", "check", source, args.Dest, "--config=rclone.conf", "--combined=errors", "--one-way"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # the command 'rclnoe check --combined' returns files that are identical, as well as the errors. So we'll remove the identical files from the errors file
+    with open("errors", "r") as input, open("temp", "w") as output:
```

Comment:
This again writes a file and reads it.  That can (and will) fail in various ways.  E.g.:

* Program doesn't run if the current directory is not writable.
* If you the directory _is_ writable but already has a file `temp`, it deletes that file.
* If filesystem is close to full (common when running on very small instances, or inside a container), it may fail for no good reason.

Can RClone just write to a pipe, which we can then read and process?

Reply:
I'll check and open an issue if necessary.